packages/server/lib/refreshTokens.ts

```diff
@@ -30,28 +30,39 @@ export function refreshTokens(): void {
 export async function exec(): Promise<void> {
     logger.info('[refreshTokens] starting');
 
-    const staleConnections = await connectionService.getOldConnections({ days: 7, limit: 500 });
+    const today = new Date();
+    const lockKey = parseInt(`${today.getFullYear()}${today.getMonth() + 1}${today.getDate()}`);
 
-    logger.info(`[refreshTokens] found ${staleConnections.length} stale connections`);
+    // Lock to prevent multiple instances of this cron job from running at the same time
+    await db.knex.transaction(async (trx) => {
+        const { rows } = await trx.raw<{ rows: { pg_try_advisory_xact_lock: boolean }[] }>(`SELECT pg_try_advisory_xact_lock(?);`, [lockKey]);
+        if (!rows || rows.length <= 0 || !rows[0]!.pg_try_advisory_xact_lock) {
```

Comment:
can probably be simplify by using optional chaining `if(!rows?.[0]?.pg_try_advisory_xact_lock)